hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -401,6 +401,16 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		})
 		return ctrl.Result{}, fmt.Errorf("failed to get config: %w", err)
 	}
+	if retryAfter != nil {
+		meta.SetStatusCondition(&nodePool.Status.Conditions, metav1.Condition{
+			Type:               hyperv1.NodePoolConfigValidConfigConditionType,
+			Status:             metav1.ConditionFalse,
+			Reason:             hyperv1.NodePoolValidationFailedConditionReason,
+			Message:            "Core ignition config has not been created yet",
+			ObservedGeneration: nodePool.Generation,
+		})
+		return reconcile.Result{RequeueAfter: *retryAfter}, nil
```

Comment:
seems like it would be simpler to just `RequeueAfter: 5 * time.Second` here and have `retryAfter` just be a `retry bool` in the return from `getConfig`.  Rather than returning functions

Reply:
Good point (although it doesn't return a function, it returns the result of a call to an inline function as the workaround for not being able to inline creation of a time.Duration pointer....)

Reply:
updated, PTAL